test/cpp/tensorexpr/test_kernel.cpp

```diff
@@ -1754,5 +1754,73 @@ TEST_F(Kernel, InputAsOutput) {
   CHECK(at::allclose(y, stack[1].toTensor()));
 }
 
+TEST_F(Kernel, ScalarOut) {
+  auto ir = R"IR(
+graph(%x : int, %y : int):
+  %z : int = aten::mul(%x, %y)
+  %r : int = aten::mul(%z, %x)
+  return (%r, %z))IR";
+  auto graph = std::make_shared<Graph>();
+  std::unordered_map<std::string, Value*> vmap;
+  parseIR(ir, graph.get(), vmap);
+  TensorExprKernel k(graph);
+
+  int64_t x = 2, y = 3, r = 0, z = 0;
+
```

Comment:
Can you also add a check here to ensure that we generate the IR as expected, namely `let` followed by a `Store`?